tests/MongoDB.Driver.Core.Tests/Core/Connections/BinaryConnection_CommandEventTests.cs

```diff
@@ -213,11 +245,25 @@ public void Should_process_a_redacted_failed_command(string commandName)
                 responseTo: requestMessage.RequestId);
             ReceiveMessages(replyMessage);
 
+            var conditionallySecuritySensitiveCommands =
+                CommandEventHelperReflector.__conditionallySecuritySensitiveCommands;
+            var commandStartedCommandShouldNotBeRedacted =
+                conditionallySecuritySensitiveCommands.ContainsKey(commandName) &&
+                !conditionallySecuritySensitiveCommands[commandName](command);
+            var commandSucceededCommandShouldNotBeRedacted =
+                conditionallySecuritySensitiveCommands.ContainsKey(commandName) &&
+                !conditionallySecuritySensitiveCommands[commandName](replyMessage.Documents[0]);
+
+            var commandStartedCommandExpectedElementCount =
+                commandStartedCommandShouldNotBeRedacted ? command.ElementCount : 0;
+            var commandSucceededCommandExpectedElementCount =
+                commandSucceededCommandShouldNotBeRedacted ? reply.ElementCount : 0;
```

Comment:
Lines 248-260 can be simplified to:

    var commandStartedCommandShouldBeRedacted = CommandEventHelperReflector.ShouldRedactMessage(commandName, command);
    var commandSucceededCommandShouldBeRedacted = CommandEventHelperReflector.ShouldRedactMessage(commandName, replyMessage.Documents[0]);
    var commandStartedCommandExpectedElementCount = commandStartedCommandShouldBeRedacted ? 0 : command.ElementCount;
    var commandSucceededCommandExpectedElementCount = commandSucceededCommandShouldBeRedacted ? 0 : reply.ElementCount;

Reply:
Done